### PR TITLE
Updated OWNERS file to include link to docs

### PR DESCRIPTION
**Pull request by rlenferink**
As the title says

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# See the OWNERS docs at https://go.k8s.io/owners
```
[the remark this answers is not in the document]
Comment (rlenferink) [reply OWNERS]: Hi @RA489 

This is not possible to do as the file is processed as a simple text file.

[the remark this answers is not in the document]
Comment (afbjorklund) [reply OWNERS]: Even though the syntax is YAML and the comment format is shell ;-)

**Pull request merged**